rpc/core/mempool.go

```diff
@@ -26,6 +26,10 @@ func BroadcastTxAsync(ctx *rpctypes.Context, tx types.Tx) (*ctypes.ResultBroadca
 	if err != nil {
 		return nil, err
 	}
+
+	env.EventBus.PublishEventPendingTx(types.EventDataTx{TxResult: types.TxResult{
```

Comment:
It is suggested to move `env.EventBus.PublishEventPendingTx` into `mempool.addTx` to promise that the tx is indeed added to mempool.

Reply:
the code has been moved into mempool.addTx()